src/gtkcompat.h

```diff
@@ -18,86 +18,16 @@
- * a plugin needs to be rebuilt and tuned to work with GTK3 we don't mind that
- * a type changes between the GTK2 and GTK3 version */
-#if ! GTK_CHECK_VERSION(3, 0, 0)
-/* undef those not to get warnings about redefinitions under GTK 2.24 */
-#	undef GTK_COMBO_BOX_TEXT
-#	undef GTK_COMBO_BOX_TEXT_CLASS
-#	undef GTK_COMBO_BOX_TEXT_GET_CLASS
-#	undef GTK_IS_COMBO_BOX_TEXT
-#	undef GTK_IS_COMBO_BOX_TEXT_CLASS
-#	undef GTK_TYPE_COMBO_BOX_TEXT
-
-#	define GTK_COMBO_BOX_TEXT					GTK_COMBO_BOX
-#	define GTK_COMBO_BOX_TEXT_CLASS				GTK_COMBO_BOX_CLASS
-#	define GTK_COMBO_BOX_TEXT_GET_CLASS			GTK_COMBO_BOX_GET_CLASS
-#	define GTK_IS_COMBO_BOX_TEXT				GTK_IS_COMBO_BOX
-#	define GTK_IS_COMBO_BOX_TEXT_CLASS			GTK_IS_COMBO_BOX_CLASS
-#	define GTK_TYPE_COMBO_BOX_TEXT				GTK_TYPE_COMBO_BOX
-#	define GtkComboBoxText						GtkComboBox
-#	define gtk_combo_box_text_new				gtk_combo_box_new_text
-#	define gtk_combo_box_text_new_with_entry	gtk_combo_box_entry_new_text
-#	define gtk_combo_box_text_append_text		gtk_combo_box_append_text
-#	define gtk_combo_box_text_insert_text		gtk_combo_box_insert_text
-#	define gtk_combo_box_text_prepend_text		gtk_combo_box_prepend_text
-#	define gtk_combo_box_text_remove			gtk_combo_box_remove_text
-#	define gtk_combo_box_text_get_active_text	gtk_combo_box_get_active_text
-#endif
-
-/* GtkWidget */
-#if ! GTK_CHECK_VERSION(3, 0, 0)
-#	define gtk_widget_get_allocated_height(widget)	(((GtkWidget *) (widget))->allocation.height)
-#	define gtk_widget_get_allocated_width(widget)	(((GtkWidget *) (widget))->allocation.width)
-#endif
-
-
-/* Mappings below only prevent some deprecation warnings on GTK3 for things
- * that didn't exist on GTK2.  That's not future-proof. */
-#if GTK_CHECK_VERSION(3, 0, 0)
-/* Gtk[VH]Box */
-#	define compat_gtk_box_new(orientation, homogeneous, spacing) \
-		((GtkWidget *)g_object_new(GTK_TYPE_BOX, \
-					 "orientation", (orientation), \
-					 "homogeneous", (homogeneous), \
-					 "spacing", (spacing), \
-					 NULL))
-#	define gtk_vbox_new(homogeneous, spacing) \
-		compat_gtk_box_new(GTK_ORIENTATION_VERTICAL, (homogeneous), (spacing))
-#	define gtk_hbox_new(homogeneous, spacing) \
-		compat_gtk_box_new(GTK_ORIENTATION_HORIZONTAL, (homogeneous), (spacing))
-/* Gtk[VH]ButtonBox */
-#	define gtk_vbutton_box_new()	gtk_button_box_new(GTK_ORIENTATION_VERTICAL)
-#	define gtk_hbutton_box_new()	gtk_button_box_new(GTK_ORIENTATION_HORIZONTAL)
-/* Gtk[VH]Separator */
-#	define gtk_vseparator_new()	gtk_separator_new(GTK_ORIENTATION_VERTICAL)
-#	define gtk_hseparator_new()	gtk_separator_new(GTK_ORIENTATION_HORIZONTAL)
-/* Gtk[VH]Paned */
-#	define gtk_vpaned_new()	gtk_paned_new(GTK_ORIENTATION_VERTICAL)
-#	define gtk_hpaned_new()	gtk_paned_new(GTK_ORIENTATION_HORIZONTAL)
-/* Gtk[VH]Scrollbar */
-#	define gtk_vscrollbar_new(adj)	gtk_scrollbar_new(GTK_ORIENTATION_VERTICAL, (adj))
-#	define gtk_hscrollbar_new(adj)	gtk_scrollbar_new(GTK_ORIENTATION_HORIZONTAL, (adj))
-#endif
```

Comment:
I think this is safe, but it might affect plugins in weird ways if they relied on weird side effects (`compat_gtk_box_new()`, or any of the other being macros for example).  It's not documented so we might just not care, but maybe it'd be worth checking if at least Geany-Plugins is fine with dropping all of it.

Reply:
Or fixing G-P plugins that do rely on undocumented features.

Reply:
Sure, but if GP does, maybe others do.

Reply:
Which is why I suggested a deprecation period.